### ISSUE TITLE
Nodetool decommission ran on the wrong node

### ISSUE THREAD

**Issue by ShlomiBalalis** [scylladb/scylla-cluster-tests#4446]
### Prerequisites

* [ ] Are you rebased to master ?
* [ ] Is it reproducible ?
* [x] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: branch-4.6
* **scylla**: 4.6.rc5

### Logs
* test_id: a1eacfb0-edb0-44d6-b900-f5bbe78ceac0
* job: https://jenkins.scylladb.com/view/scylla-4.6/job/scylla-4.6/job/longevity/job/longevity-50gb-3days-arm-test/2/

### Description

At `2022-02-09 00:38:43.924` a RunUniqueSequence nemesis has started, which consists of manager repair, terminate and replace and grow shrink cluster nemeses.

At `2022-02-09 02:00:22,818` a nodetool decommission has started (as part of the grow shrink nemesis) with node 8 being the target node:
```
< t:2022-02-09 02:00:22,818 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "/usr/bin/nodetool -u cassandra -pw 'cassandra'  decommission "...
< t:2022-02-09 02:00:22,819 f:file_logger.py  l:89   c:sdcm.sct_events.file_logger p:INFO  > 2022-02-09 02:00:22.817: (InfoEvent Severity.NORMAL) period_type=not-set event_id=ae606dae-9722-4868-aacf-35090c0ed9f7: message=StartEvent - ShrinkCluster started decommissioning a node Node longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-8 [18.212.108.46 | 10.0.14.67] (seed: False)
```
We specified no other node in the command, so, by default, the node which the command was executed on should have decommissioned, i.e. node 8 (18.212.108.46 | 10.0.14.67). However, it was node 1 (34.201.130.182 | 10.0.12.142) that was decommissioned:
```
2022-02-09T02:00:26+00:00 longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-1 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: starts
2022-02-09T02:00:26+00:00 longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-1 !    INFO |  [shard 0] storage_service - decommission[c984326b-fe0f-43c1-87f7-f342cabe119e]: Started decommission operation, removing node=10.0.12.142, sync_nodes=[10.0.14.35, 10.0.15.181, 10.0.15.241, 10.0.12.94, 10.0.12.142, 10.0.15.152, 10.0.14.67], ignore_nodes=[]
2022-02-09T02:00:26+00:00 longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-1 !    INFO |  [shard 0] storage_service - decommission[c984326b-fe0f-43c1-87f7-f342cabe119e]: Added node=10.0.12.142 as leaving node, coordinator=10.0.12.142
2022-02-09T02:00:29+00:00 longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-1 !    INFO |  [shard 0] storage_service - decommission[c984326b-fe0f-43c1-87f7-f342cabe119e]: Started heartbeat_updater
2022-02-09T02:00:29+00:00 longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-1 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: unbootstrap starts
```

from node 8:
```
2022-02-09T02:00:26+00:00 longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-8 !    INFO |  [shard 0] storage_service - decommission[c984326b-fe0f-43c1-87f7-f342cabe119e]: Added node=10.0.12.142 as leaving node, coordinator=10.0.12.142
```
```
< t:2022-02-09 02:08:37,832 f:cluster.py      l:2611 c:sdcm.cluster_aws     p:DEBUG > Node longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-8 [18.212.108.46 | 10.0.14.67] (seed: False): Command '/usr/bin/nodetool -u cassandra -pw 'cassandra'  decommission ' duration -> 495.0104019519931 s
```

I believe that sct accidently targeted the wrong node in the nemesis, which caused this mishap.
However, this mishap doesn't appear to be consistent, seeing that there were a few decommissions in the run, and it doesn't appear that any of them suffered the same fate.

**Expected behavior:** The target node should be decommissioned

**Actual behavior:** A different node was decommissioned

Node list:
```
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-1 [34.201.130.182 | 10.0.12.142] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-2 [34.227.84.82 | 10.0.15.201] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-3 [3.90.242.245 | 10.0.15.104] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-4 [52.90.25.236 | 10.0.13.135] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-5 [52.91.29.106 | 10.0.15.242] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-6 [52.23.215.187 | 10.0.13.113] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-7 [54.145.98.20 | 10.0.14.72] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-8 [18.212.108.46 | 10.0.14.67] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-9 [3.81.104.83 | 10.0.14.3] (seed
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-10 [3.90.240.89 | 10.0.15.237] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-11 [3.84.129.123 | 10.0.15.218]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-12 [52.90.239.100 | 10.0.14.131] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-13 [35.175.231.161 | 10.0.12.21] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-14 [204.236.193.2 | 10.0.12.196] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-15 [34.230.13.106 | 10.0.15.194] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-16 [54.197.128.104 | 10.0.14.151]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-17 [18.215.231.6 | 10.0.15.152]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-18 [54.166.183.51 | 10.0.12.94]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-19 [52.90.88.167 | 10.0.15.241]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-20 [50.16.168.248 | 10.0.12.83]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-21 [52.207.158.237 | 10.0.15.181]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-22 [18.234.23.224 | 10.0.14.35]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-23 [54.80.155.221 | 10.0.14.239] 
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-24 [34.228.224.204 | 10.0.13.134]
longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-25 [3.85.8.139 | 10.0.12.7]
```

Logs:
```
+--------------------------------------------------------------------------------------------------------------------------------------------------+
|                                         Collected logs by test-id: a1eacfb0-edb0-44d6-b900-f5bbe78ceac0                                          |
+-------------+------------------------------------------------------------------------------------------------------------------------------------+
| Cluster set | Link                                                                                                                               |
+-------------+------------------------------------------------------------------------------------------------------------------------------------+
| db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/db-cluster-a1eacfb0.tar.gz     |
| loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/loader-set-a1eacfb0.tar.gz     |
| normal      | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/normal-a1eacfb0.log.tar.gz     |
| summary     | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/summary-a1eacfb0.log.tar.gz    |
| events      | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/events-a1eacfb0.log.tar.gz     |
| output      | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/output-a1eacfb0.log.tar.gz     |
| debug       | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/debug-a1eacfb0.log.tar.gz      |
| sct         | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/sct-a1eacfb0.log.tar.gz        |
| error       | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/error-a1eacfb0.log.tar.gz      |
| critical    | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/critical-a1eacfb0.log.tar.gz   |
| raw_events  | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/raw_events-a1eacfb0.log.tar.gz |
| warning     | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/warning-a1eacfb0.log.tar.gz    |
| argus       | https://cloudius-jenkins-test.s3.amazonaws.com/a1eacfb0-edb0-44d6-b900-f5bbe78ceac0/20220213_085624/argus-a1eacfb0.log.tar.gz      |
+-------------+------------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by fgelcer** [comment]
@ShlomiBalalis , it is intentional (when the nemesis is `ShrinkCluster`), as you can see in here:
https://github.com/scylladb/scylla-cluster-tests/blob/92f878e26f90055b343a1ea69186596d94a8359d/sdcm/nemesis.py#L298

that comes from https://github.com/scylladb/scylla-cluster-tests/blob/92f878e26f90055b343a1ea69186596d94a8359d/sdcm/nemesis.py#L2732

and it is called here:
https://github.com/scylladb/scylla-cluster-tests/blob/92f878e26f90055b343a1ea69186596d94a8359d/sdcm/nemesis.py#L2797

it means that, because in this nemesis we have the ability to run multiple `grow_cluster` (and the same amount of `shrink_cluster`), it cannot run on the very same "`target_node`" set in the "main" nemesis...

so it is working as expected.

**Comment by ShlomiBalalis** [comment]
Yes, but it was speccifically specified that node 8 is to be removed: 
`< t:2022-02-09 02:00:22,816 f:nemesis.py      l:2732 c:sdcm.nemesis         p:INFO  > sdcm.nemesis.SisyphusMonkey: Next node will be removed Node longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-8 [18.212.108.46 | 10.0.14.67] (seed: False)`

**Comment by fgelcer** [comment]
i see the last `grow_shrink_cluster` failed because `node-8` was not decommissioned... need to investigate further..
@ShlomiBalalis , please add here more evidence that the wrong node was decommissioned (each decommission has a UUID) in the same decommission process... 

maybe it "retry" because the original decommission of `node-8` failed, and then it leave the cluster with the same amount of nodes it decommissioned another node?

**Comment by fgelcer** [comment]
i see the last `grow_shrink_cluster` failed because `node-8` was not decommissioned... need to investigate further..
@ShlomiBalalis , please add here more evidence that the wrong node was decommissioned (each decommission has a UUID) in the same decommission process... 

maybe it "retry" because the original decommission of `node-8` failed, and then it leave the cluster with the same amount of nodes it decommissioned another node?

**Comment by ShlomiBalalis** [comment]
> maybe it "retry" because the original decommission of `node-8` failed, and then it leave the cluster with the same amount of nodes it decommissioned another node?

In this context, who is retrying? SCT or scylla? Cause I'm sure SCT didn't retry since I see no option for it to happen in the function. And if you mean scylla then this is the first and only decommission I saw in the log for this timeframe, and ever since it started node one was the leaving node:
`2022-02-09T02:00:26+00:00 longevity-tls-50gb-3d-4-6-db-node-a1eacfb0-8 !    INFO |  [shard 0] storage_service - decommission[c984326b-fe0f-43c1-87f7-f342cabe119e]: Added node=10.0.12.142 as leaving node, coordinator=10.0.12.142`
